File: algorithms/sleep_sort/java/sleep_sort.java
public class Sleepsort {

    private static int[] inputArray = { 3, 1, 2, 1, 181, 10};

    public static void main(String[] args) throws InterruptedException {
        Sleepsort ss = new Sleepsort();
        ss.sleepsort(inputArray);
    }

    public void sleepsort(int[] array) throws InterruptedException {
        for (int val : array) {
            Thread thread = new Thread(new SleepsortThread(val));
            thread.start();
        }
    }
}

class SleepsortThread implements Runnable {
    private int val;

    public SleepsortThread(int val) {
        this.val = val;
    }

    @Override
    public void run() {
        try {
            Thread.sleep(val);
            System.out.println(val);
        } catch (InterruptedException e) {
            // Oops...
        }
    }
}

File: algorithms/sleep_sort/python/sleepsort.py
from threading import Thread
from time import sleep

def sort_worker(arg):
    sleep(arg)
    print str(arg)+ ', ',

if __name__ == "__main__":
    threads = []
    numbers = [1,4,2,5,6,7,8,4,8,10,9]

    for number in numbers:
        new_thread = Thread(target=sort_worker, args=(number,))
        new_thread.start()
        threads.append(new_thread)

    for thread in threads:
        thread.join()